lang_GENERIC/analyze/AST_to_IL.ml

```diff
@@ -71,16 +71,40 @@ let sgrep_construct any_generic =
   error_any any_generic "Sgrep Construct"
 (*e: function [[AST_to_IL.sgrep_construct]] *)
 
-(*s: function [[AST_to_IL.todo]] *)
-let todo any_generic =
-  error_any any_generic "TODO Construct"
-(*e: function [[AST_to_IL.todo]] *)
-
 (*s: function [[AST_to_IL.impossible]] *)
 let impossible any_generic =
   error_any any_generic "Impossible Construct"
 (*e: function [[AST_to_IL.impossible]] *)
 
+exception Todo of G.any
+
+(*s: function [[AST_to_IL.todo]] *)
+let todo gany =
+  raise (Todo gany)
+(*e: function [[AST_to_IL.todo]] *)
+
+let todo_warning gany =
+  let toks = Lib_AST.ii_of_any gany in
+  let s = AST_generic.show_any gany in
+  let msg = spf "[DFA] Unsupported construct:\n<-----\n%s\n----->" s in
```

Comment:
what does DFA mean?

Reply:
DFA stands for Data-Flow Analysis. I actually wanted some feedback from you regarding the warnings. Right now these warnings will be rather verbose for a normal user, maybe the user should just get a brief "unsupported construct will affect precision/recall of dataflow analyses", and the verbose warning could be logged somewhere else (or in verbose mode). I haven't looked very careful into the different logging/verbosity options of Pfff.

Reply:
I kept the warning message very simple, it simply informs about an unsupported construct and points to the source location. That should be enough to identify the construct in many cases. We could improve on that later on if needed.